File: doc/_i18n/en/tutorials/introduction/multi-robot-controller/logic_setup.cpp
// In the header
enum DoorPhase
{
      APPROACH = 0,
      HANDLE,
      OPEN
};
// A private property of our controller
DoorPhase phase = APPROACH;
// A new method for our controller
void switch_phase()
{
  if(phase == APPROACH && 0 /** we write this condition later */)
  {
    /** Setup the HANDLE phase */
    phase = HANDLE;
  }
  else if(phase == HANDLE && 0 /** we write this condition later */)
  {
    /** Setup the OPEN phase */
    phase = OPEN;
  }
}
// Call this in the run function
bool MyFirstController::run()
{
  switch_phase();
  return mc_control::MCController::run();
}

File: doc/_i18n/en/tutorials/introduction/multi-robot-controller/logic_setup.py
# Declare constants
APPROACH = 0
HANDLE = 1
OPEN = 2
# In constructor
self.phase = APPROACH
# New method for our controller
def switch_phase(self):
  if self.phase == APPROACH and False: # We write this condition later
    # Setup the HANDLE phase
    self.phase = HANDLE
  elif self.phase == HANDLE and False: # We write this condition later
    # Setup the OPEN phase
    self.phase = OPEN
# Call this in the run callback
def run_callback(self):
    self.switch_phase()
    return True
